python/ray/_private/worker.py

```diff
@@ -125,7 +125,40 @@
 DAGNode = TypeVar("DAGNode")
 
 
-class RemoteFunctionNoArgs(Generic[R]):
+# Only used for type annotations as a placeholder
+Undefined: Any = object()
+
+
+# TypeVar for self-referential generics in `RemoteFunction[N]`.
+RF = TypeVar("RF", bound="HasOptions")
+
+
+class HasOptions(Protocol):
```

Comment:
I understand that generally it is better to have proper type annotations. Just wondering in this particular case, are you trying to achieve something specific? What is the use case that you were trying to enable? IIUC currently Ray runs mypy on a limited number of files.